File: Client_Root/Client/Assets/Scripts/Network/Messages/ToRoom/PreparationStateToR.cs
using FlatBuffers;

public class PreparationStateToR : IMessage
{
    public const ushort MESSAGE_ID = MessageID.PreparationStateToR_ID;

    public float m_fState;

    public override ushort GetID()
    {
        return MESSAGE_ID;
    }

    public override IMessage Clone()
    {
        return null; 
    }

    public override byte[] Serialize()
    {
        FBS.PreparationStateToR.StartPreparationStateToR(m_Builder);
        FBS.PreparationStateToR.AddState(m_Builder, m_fState);
        var data = FBS.PreparationStateToR.EndPreparationStateToR(m_Builder);

        m_Builder.Finish(data.Value);

        return m_Builder.SizedByteArray();
    }

    public override bool Deserialize(byte[] bytes)
    {
        var buf = new ByteBuffer(bytes);

        var data = FBS.PreparationStateToR.GetRootAsPreparationStateToR(buf);

        m_fState = data.State;

        return true;
    }
}

File: Server_Root/Server/RoomServer/Messages/ToRoom/PreparationStateToR.cpp
#include "stdafx.h"
#include "PreparationStateToR.h"
#include "../../../FBSFiles/PreparationStateToR_generated.h"

PreparationStateToR::PreparationStateToR()
{
}

PreparationStateToR::~PreparationStateToR()
{
}

unsigned short PreparationStateToR::GetID()
{
	return MESSAGE_ID;
}

IMessage* PreparationStateToR::Clone()
{
	return NULL;
}

const char* PreparationStateToR::Serialize(int* pLength)
{
	FBS::PreparationStateToRBuilder data_builder(m_Builder);
	data_builder.add_State(m_fState);
	auto data = data_builder.Finish();

	m_Builder.Finish(data);

	*pLength = m_Builder.GetSize();

	return (char*)m_Builder.GetBufferPointer();
}

bool PreparationStateToR::Deserialize(const char* pChar)
{
	auto data = flatbuffers::GetRoot<FBS::PreparationStateToR>((const void*)pChar);

	m_fState = data->State();

	return true;
}